core/src/main/java/org/elasticsearch/common/settings/SettingsModule.java

```diff
@@ -71,19 +71,22 @@ protected void configure() {
      * the setting during startup.
      */
     public void registerSetting(Setting<?> setting) {
-        switch (setting.getScope()) {
-            case CLUSTER:
-                if (clusterSettings.containsKey(setting.getKey())) {
-                    throw new IllegalArgumentException("Cannot register setting [" + setting.getKey() + "] twice");
-                }
-                clusterSettings.put(setting.getKey(), setting);
-                break;
-            case INDEX:
-                if (indexSettings.containsKey(setting.getKey())) {
-                    throw new IllegalArgumentException("Cannot register setting [" + setting.getKey() + "] twice");
-                }
-                indexSettings.put(setting.getKey(), setting);
-                break;
+        if (setting.isFiltered()) {
+            if (settingsFilterPattern.contains(setting.getKey()) == false) {
+                registerSettingsFilter(setting.getKey());
+            }
+        }
+        if (setting.hasClusterScope()) {
+            if (clusterSettings.containsKey(setting.getKey())) {
+                throw new IllegalArgumentException("Cannot register setting [" + setting.getKey() + "] twice");
+            }
+            clusterSettings.put(setting.getKey(), setting);
+        }
+        if (setting.hasIndexScope()) {
+            if (indexSettings.containsKey(setting.getKey())) {
+                throw new IllegalArgumentException("Cannot register setting [" + setting.getKey() + "] twice");
+            }
+            indexSettings.put(setting.getKey(), setting);
```

Comment:
I think that is problematic, lets check this here where it's relevant not in the setting itself. Lets allow no properties otherwise the var-args are broken IMO